lib/standards/tests/fieldsMustHaveLabelsOrTitles.js

```diff
@@ -5,34 +5,80 @@ module.exports = {
 
   failsForEach: [
     'visible input element (<input>, <textarea> and <select>) with no title ' +
-    "attribute or <label> element referring to it (in the label's for attribute)",
+      "attribute or <label> element referring to it (in the label's for attribute)",
     '<button> element with no text (or only whitespace as text)'
   ],
 
   test: function ({ $, fail }) {
-    $("input:visible:not([title], [type='hidden'], [type='submit'], [type='reset'], [type='image']), " +
-      'textarea:visible:not([title]), select:visible:not([title])')
-      .filter(function (index, field) {
-        return !hasIdOrLabel($(field))
-      })
-      .each(function (index, element) {
-        fail('Field has no label or title attribute:', element)
-      })
+    $(
+      "input:visible:not([title], [type='hidden'], [type='submit'], [type='reset'], [type='image']), " +
+        'textarea:visible:not([title]), select:visible:not([title])'
+    ).each(function (index, field) {
+      const $field = $(field)
+
+      if (!hasId($field) || !hasLabel($field)) {
+        fail('Field has no label or title attribute:', field)
+      }
+    })
+
+    $('input[type="image"]').each(function (index, field) {
+      const fieldSrc = $(field).attr('src')
+      const fieldAlt = $(field).attr('alt')
+
+      if (Boolean(fieldSrc && fieldSrc.trim()) === false) {
+        fail('Field with type of image has no src:', field)
+      }
+      if (Boolean(fieldAlt && fieldAlt.trim()) === false) {
+        fail('Field with type of image has no alt text:', field)
+      }
+    })
 
     $('button')
       .filter(function (index, button) {
-        return $(button).text().trim() === ''
+        const buttonHasNoText = $(button).text().trim() === ''
+        return buttonHasNoText
+      })
+
+      .filter(function (index, button) {
+        const buttonHasNoAriaLabel = hasAriaLabel($(button)) === false
+        return buttonHasNoAriaLabel
+      })
+      .filter(function (index, button) {
+        const buttonHasNoAriaLabelledBy =
+          hasAriaLabelledBy($(button)) === false
```

Comment:
If you make `hasAriaLabelledBy` take the properties that filter calls a function with, this could be simplified to:
`.filter(!hasAriaLabelledBy)`.
Same with the other simple filters. Might read a little nicer.